wazo_webhookd/services/mobile/plugin.py

```diff
@@ -0,0 +1,213 @@
+                                              callback=self.on_external_auth_deleted)
+        logger.info('Mobile push notification plugin is started')
+
+    def on_external_auth_added(self, body, event):
+        if body['data'].get('external_auth_name') == 'mobile':
+            user_uuid = body['data']['user_uuid']
+            # TODO(sileht): Should come with the event
+            tenant_uuid = self.get_tenant_uuid(user_uuid)
+            self.subscription_service.create({
+                'name': ('Push notification mobile for user '
+                         '{}/{}'.format(tenant_uuid, user_uuid)),
+                'service': 'mobile',
+                'events': [
+                    'call_created',
+                    'chatd_user_room_message_created',
+                    'call_push_notification',
+                    'user_voicemail_message_created'
+                ],
+                'events_user_uuid': user_uuid,
+                # 'events_tenant_uuid': tenant_uuid,
+                'owner_user_uuid': user_uuid,
+                'owner_tenant_uuid': tenant_uuid,
+                'config': {},
+                'metadata': {'mobile': 'true'},
+            })
+            logger.info('User registered: %s/%s', tenant_uuid, user_uuid)
+
+    def on_external_auth_deleted(self, body, event):
+        if body['data'].get('external_auth_name') == 'mobile':
+            user_uuid = body['data']['user_uuid']
+            # TODO(sileht): Should come with the event
+            tenant_uuid = self.get_tenant_uuid(user_uuid)
+            subscriptions = self.subscription_service.list(
+                owner_user_uuid=user_uuid,
+                owner_tenant_uuids=[tenant_uuid],
+                search_metadata={'mobile': 'true'},
+            )
+            for subscription in subscriptions:
+                self.subscription_service.delete(subscription.uuid)
+            logger.info('User unregistered: %s/%s', tenant_uuid, user_uuid)
+
+    def get_tenant_uuid(self, user_uuid):
+        auth = self.get_auth(self._config)
+        return auth.users.get(user_uuid)["tenant_uuid"]
+
+    @classmethod
+    def get_auth(cls, config):
+        auth_config = dict(config['auth'])
+        # FIXME(sileht): Keep the certificate
+        auth_config['verify_certificate'] = False
+        auth = AuthClient(**auth_config)
+        token = auth.token.new('wazo_user', expiration=3600)
+        auth.set_token(token["token"])
+        auth.username = None
+        auth.password = None
+        return auth
+
+    @classmethod
+    def get_external_token(cls, config, user_uuid):
+        auth = cls.get_auth(config)
+        token = auth.external.get('mobile', user_uuid)
+        tenant_uuid = auth.users.get(user_uuid)['tenant_uuid']
+        external_config = auth.external.get_config('mobile', tenant_uuid)
+
+        if token["apns_token"]:
+            external_config['ios_apns_cert'] = '/tmp/ios.pem'
```

Comment:
All tenants use the same file `/tmp/ios.pem`

which could mix the values for different tenants when using the value stored in the file